pkg/app/app.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package app
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/subnet-evm/core"
+)
+
+const (
+	WriteReadReadPerms = 0o644
+)
+
+var errChainIDExists = errors.New("The provided chain ID already exists! Try another one")
+
+type Avalanche struct {
+	Log     logging.Logger
+	baseDir string
+}
+
+func New(baseDir string, log logging.Logger) *Avalanche {
+	return &Avalanche{
+		baseDir: baseDir,
+		Log:     log,
+	}
+}
+
+func (app *Avalanche) GetBaseDir() string {
+	return app.baseDir
+}
+
+func (app *Avalanche) GetGenesisPath(subnetName string) string {
+	return filepath.Join(app.baseDir, subnetName+constants.Genesis_suffix)
+}
+
+func (app *Avalanche) GetSidecarPath(subnetName string) string {
+	return filepath.Join(app.baseDir, subnetName+constants.Sidecar_suffix)
+}
+
+func (app *Avalanche) WriteGenesisFile(subnetName string, genesisBytes []byte) error {
```

Comment:
I'm glad we moved this functionality out of `cmd`. I was running into problems because I couldn't import these tools due to a cyclical import.